### DISCUSSION TITLE
organize os module

### DISCUSSION THREAD

**Discussion by bokunodev** [vlang/v#18642] [Ideas]
### Describe the feature

many functions in os module are inconsistent and many functions also missing like stat, lstat etc.

`file_last_mod_unix` does not return an error and does not have `fd` variant which accept file descriptor instead of path.

https://github.com/vlang/v/blob/a9a04bba55742f333db18ce39e00dda4c6e4c55f/vlib/os/os.c.v#L940

### Use Case

i take a look at v script or v for bash user. it suggests to use v (or vsh, idk the differ) as cross platform shell script alternative.
but so many basic function as missing from stdlib and os module is full of mess.

### Proposed Solution

rework os module
1. shorten fn names that are too long (`stat` instead of `file_last_mod_unix`)
3. add more relevant method to `os.File` (instead of having alternative fn that work on fd, add it as method to `os.File` instead)
4. make all `fn`s to either return error or value (so many unsafe blocks without proper error handling in os package.)
5. move path related `fn`s into separate module (path manipulation related `fn` should have their own module)

### Acknowledgements

- [X] This feature might incur a breaking change
- [x] I may be able to implement this feature request

### Version used

V 0.3.2 a9a04bb

### Environment details (OS name and version, etc.)

Linux local.pc 6.1.6-artix1-1 #1 SMP PREEMPT_DYNAMIC Thu, 19 Jan 2023 21:37:40 +0000 x86_64 GNU/Linux

**Comment by ylluminate** [comment]
What you seem to be suggesting here is to bring [`vlib/os`](https://github.com/vlang/v/tree/master/vlib/os) into a kind of parity with https://github.com/golang/go/tree/master/src/os, right?

**Comment by bokunodev** [comment]
Go is a good example, yes.
(maybe, once the compiler is more mature)
